tests/functional/restart/56-db-saves-type.t

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test restarting a simple workflow with a task still running (orphaned)
+. "$(dirname "$0")/test_header"
+set_test_number 3
+init_workflow "${TEST_NAME_BASE}" <<'__FLOW_CONFIG__'
+#!Jinja2
+
+{{ assert(a_str is string, 'variable "a_str" was not a string') }}
+{{ assert(an_int is integer, 'variable "an_int" was not an int') }}
+{{ assert(a_float is float, 'variable "a_float" was not a float') }}
+{{ assert(a_dict is mapping, 'variable "a_dict" was not a mapping') }}
```

Comment:
(unresolving because this applies to the earlier examples too)

Reply:
having gone to the bother of setting `a_float = 1.1111` why not check `a_float == 1.1111`?

Reply:
I can't get the above working - does Jinja2 support set properly?

Reply:
> I can't get the above working - does Jinja2 support set properly?

Try the Jinja2 docs, maybe something like `set(1, 2)`.

It doesn't matter whether or not Jinja2 supports defining sets using the `{% set` command since we don't use the `{% set` command, we pass the value to Jinja2 straight from Python.

So long as literal_eval/repr support a type all good:

```python
from ast import literal_eval

for value in [ 
    1,  
    1.1,
    True,
    None,
    [1,],  
    {1,},
    (1,),
    {'a': 1},
]:        
    assert literal_eval(repr(value)) == value
```